### PR TITLE
Add FindROOT and Geometry to CMake, change some pointers to smart pointers

### PR DESCRIPTION
**Pull request by grey277**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -21,8 +22,10 @@ int main(int argc, char** argv)
   DATE_AND_TIME();
   INFO("J-PET Event Display started");
   INFO("*********************");
-  TRint* theApp = new TRint("App", &argc, argv);
-  TCanvas* canvas = new TCanvas("MyCanvas", "Test Canvas", 10, 10, 900, 500);
+  std::shared_ptr<TRint> theApp = std::make_shared<TRint>("App", &argc, argv);
```
Comment (wkrzemien) [inline main.cpp]: Probably should be unique pointer, not the shared one, 
at least for TRint

File: src/GeometryVisualizator.h [hunk 2]
```diff
@@ -25,35 +25,38 @@
...
-  std::string getLayerNodeName(int layer) const;
-  std::string getStripNodeName(int strip) const;
-
-private:
-  enum ColorTable { kBlack = 1, kRed = 2, kBlue = 34, kGreen = 30};
-  TGeoManager* fGeoManager;
-  TCanvas* fMyCanv;
-
-};
+  using namespace std;
```
Comment (wkrzemien) [inline src/GeometryVisualizator.h]: using namespace in a header file is a bad practice.. It leads to namespace pollution  cause every file that includes your header gets automatically included all things from this namespace. That should be avoided.  See e.g. 
Eckel or Alesanderscu or many others.

i

File: src/GeometryVisualizator.h [hunk 3]
```diff
@@ -25,35 +25,38 @@
...
+    void drawStrips(const std::map<int, std::vector<int> >& selection);
+    void drawPads();
+    void setAllStripsUnvisible();
+    void setVisibility(const std::map<int, std::vector<int> >& selection);
+    std::string getLayerNodeName(int layer) const;
+    std::string getStripNodeName(int strip) const;
+
+  private:
+    enum ColorTable { kBlack = 1, kRed = 2, kBlue = 34, kGreen = 30};
+    shared_ptr<TGeoManager> fGeoManager;
```
Comment (wkrzemien) [inline src/GeometryVisualizator.h]: by default should be unique pointer unless you know that you want shared pointer

File: src/GeometryVisualizator.h [hunk 4]
```diff
@@ -25,35 +25,38 @@
...
-};
+  using namespace std;
+
+  class GeometryVisualizator
+  {
+  public:
+    GeometryVisualizator(shared_ptr<TCanvas> canv);
+    ~GeometryVisualizator();
+    bool isGeoManagerInitialized() const;
+    void loadGeometry(TString geomFile);
```
Comment (wkrzemien) [inline src/GeometryVisualizator.h]: If possibel change TString to const std::string&

**Comment by grey277** [comment]
Add updated files, in loadGeometry it is needed to cast std::string to TString to create TFile

**Pull request merged**